src/tree/index.tsx

```diff
@@ -0,0 +1,400 @@
+import { create, tsx, diffProperty } from '@dojo/framework/core/vdom';
+import { RenderResult } from '@dojo/framework/core/interfaces';
+import { createICacheMiddleware } from '@dojo/framework/core/middleware/icache';
+import { createResourceMiddleware } from '@dojo/framework/core/middleware/resources';
+import theme from '@dojo/framework/core/middleware/theme';
+import focus from '@dojo/framework/core/middleware/focus';
+import { flat } from '@dojo/framework/shim/array';
+
+import { Keys } from '../common/util';
+import Icon from '../icon';
+import Checkbox from '../checkbox';
+import { ListItem } from '../list';
+
+import * as css from '../theme/default/tree.m.css';
+import LoadingIndicator from '../loading-indicator';
+
+export interface TreeNodeOption {
+	id: string;
+	parent: string;
+	value: string;
+	hasChildren: boolean;
+}
+
+export interface TreeProperties {
+	checkable?: boolean;
+	selectable?: boolean;
+	checkedNodes?: string[];
+	expandedNodes?: string[];
+	selectedNode?: string;
+	disabledNodes?: string[];
+	parentSelection?: boolean;
+	initialExpanded?(id: string): void;
```

Comment:
Why are initialExpanded and initialChecked functions when the controlled versions of those are simply string arrays? They should both be string arrays.